README.md

```diff
@@ -1,6 +1,15 @@
 # Powertools for AWS Lambda (TypeScript) <!-- omit in toc -->
 
-[![Join our Discord](https://dcbadge.vercel.app/api/server/B8zZKbbyET)](https://discord.gg/B8zZKbbyET)
+[![Build](https://github.com/aws-powertools/powertools-lambda-typescript/actions/workflows/pr-run-linting-check-and-unit-tests.yml/badge.svg)]([https://github.com/aws-powertools/powertools-lambda-typescript/actions/workflows/pr-run-linting-check-and-unit-tests.yml])
```

Comment:
I would also remove this one, and rename back to the original name the workflow.

The workflow linked here is the one that runs on new PRs or when new code is pushed to an existing PR. The one that is run when a PR is merged would be [this other](https://github.com/aws-powertools/powertools-lambda-typescript/blob/01d3dc4f0e25edbf1cc86e34b09e1a9fbc4b281d/.github/workflows/on_merged_pr.yml) one.


Overall I think this is a bit redundant anyway since the same info on the last merged changes on `main` is right next to the files:
<img width="625" alt="Screenshot 2024-02-14 at 19 24 17" src="https://github.com/aws-powertools/powertools-lambda-typescript/assets/7353869/9501a8d1-45b8-49a4-8aee-e933eb2887bd">

Reply:
Imho it does not hurt to have redundant information, there are no costs for having this badge. Overall, we have many places where the information is redundant, and it's often desirable. 

Regarding the name, I think we should go a step further and rename the workflow to "Quality check", to abstract the steps, instead of `lint-unit-e2e-fuzzy-special-checks`. Then we can keep it for both PR merge semantic and the badge.

Reply:
OK, but it's not just about the name.

The workflow linked in this badge runs on PR so showing the badge doesn't make sense since it'll show the status of the last check ran on any of the open PRs.

If we want to add the badge, then we need link it to a workflow that runs on the latest code currently on `main`.

Reply:
> If we want to add the badge, then we need link it to a workflow that runs on the latest code currently on main.

Ah, I missed this point, sorry. You are right, we should only include the latest merge to main.

Reply:
Np, I was also talking about two things in the same comment and could've been clearer.

So I'm on board with adding the badge and using a more meaningful name like you suggested, however I think we should address this in a separate PR.

After looking at the workflows again, we don't really have a true "code quality" workflow that runs on merged PRs - which is a problem on its own.

To verify this, I just merged this Dependabot PR, and these below were the workflows that run as a result:
<img width="1159" alt="image" src="https://github.com/aws-powertools/powertools-lambda-typescript/assets/7353869/671ee586-3392-4856-b68c-a9498825a316">

- Scorecard
- Release drafter
- Record PR details + label related issue with `status/pending-release`

The Python repo on the other hand, has a workflow called `quality-check.yml` ([here](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/workflows/quality_check.yml)) that runs both on PRs and `push` events (aka merge commits on `main` in our case).

The closest thing we have, is the "On PR update" workflow ([here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/workflows/pr-run-linting-check-and-unit-tests.yml#L3-L5)), that in turn calls the test & lint reusable workflow.

I don't have a strong opinion on whether we should modify the "On PR update" workflow to behave like the "Quality Check" in Python, or leave that alone & add our own version of "Quality Check" that runs on `push` events on `main`. I also don't know if we should run integration tests whenever we merge on `main` like Python does (possibly yes?).

I do think however this is a larger discussion and in the interest of keeping the scope of this PR focused, I suggest we remove this one badge from this PR and open a dedicated issue that we can pick up right away after merging this PR.

Once we have the appropriate workflows in place, they're documented, and know they're working consistently then we can add the badge back to the README.

What do you both think?

Reply:
I agree, this needs a bigger discussion that is outside the scope of this PR. Will remove the badge and undo the workflow rename